lib/MailNotifications.php

```diff
@@ -0,0 +1,330 @@
+	 * @return ?IMessage message contents
+	 */
+	protected function prepareEmailMessage(string $uid, array $notifications, string $language): ?IMessage {
+		$user = $this->userManager->get($uid);
+		$userEmailAddress = $user->getEMailAddress();
+
+		if (empty($userEmailAddress)) {
+			return null;
+		}
+
+		// Prepare our email template
+		$l10n = $this->l10nFactory->get('notifications', $language);
+
+		$template = $this->mailer->createEMailTemplate('notifications.EmailNotification', [
+			'displayname' => $user->getDisplayName(),
+			'url' => $this->urlGenerator->getAbsoluteURL('/')
+		]);
+
+		// Prepare email header
+		$template->addHeader();
+		$template->addHeading($l10n->t('Hello %s',[$user->getDisplayName()]), $l10n->t('Hello %s,',[$user->getDisplayName()]));
+
+		// Prepare email subject and body mentioning amount of notifications
+		$homeLink = '<a href="' . $this->urlGenerator->getAbsoluteURL('/') . '">' . htmlspecialchars($this->defaults->getName()) . '</a>';
+		$notificationsCount = count($notifications);
+		$template->setSubject($l10n->n('New notification for %s', '%n new notifications for %s', $notificationsCount, [$this->defaults->getName()]));
+		$template->addBodyText(
+			$l10n->n('You have a new notification for %s', 'You have %n new notifications for %s', $notificationsCount, [$homeLink]),
+			$l10n->n('You have a new notification for %s', 'You have %n new notifications for %s', $notificationsCount, [$this->urlGenerator->getAbsoluteURL('/')])
+		);
+
+		// Prepare email body with the content of missed notifications
+		// Notifications are assumed to be passed in in descending order (latest first). Reversing to present chronologically.
+		$notifications = array_reverse($notifications);
+
+		$timezone = $this->config->getUserValue($uid, 'core', 'timezone', date_default_timezone_get());
+
+		foreach ($notifications as $notification) {
+			$relativeDateTime = $this->dateFormatter->formatDateTimeRelativeDay($notification->getDateTime(), 'long', 'short', new \DateTimeZone($timezone), $l10n);
+			$template->addBodyListItem($this->getHTMLContents($notification), $relativeDateTime, $notification->getIcon(), $notification->getParsedSubject());
+
+			// Buttons probably were not intended for this, but it works ok enough for showing the idea.
+			$actions = $notification->getParsedActions();
+			foreach ($actions as $action) {
+				if ($action->getRequestType() === IAction::TYPE_WEB) {
+					$template->addBodyButton($action->getLabel(), $action->getLink());
+				}
+			}
+		}
+
+		// Prepare email footer
+		$template->addBodyText(
+			$l10n->t('You can change the frequency of these emails or disable them <a href="%s">here</a>.', $this->urlGenerator->linkToRouteAbsolute('settings.PersonalSettings.index', ['section' => 'notifications'])),
+			$l10n->t('You can change the frequency of these emails or disable them here: %s', $this->urlGenerator->linkToRouteAbsolute('settings.PersonalSettings.index', ['section' => 'notifications']))
```

Comment:
```suggestion
			$l10n->t('You can change the frequency of these emails or disable them in the <a href="%s">settings</a>.', $this->urlGenerator->linkToRouteAbsolute('settings.PersonalSettings.index', ['section' => 'notifications'])),
			$l10n->t('You can change the frequency of these emails or disable them in the settings: %s', $this->urlGenerator->linkToRouteAbsolute('settings.PersonalSettings.index', ['section' => 'notifications']))
```

"here" is a bad word for links when being read by screenreaders